tool/internal/ast/parser.go

```diff
@@ -49,6 +49,21 @@ func (ap *AstParser) Parse(filePath string, mode parser.Mode) (*dst.File, error)
 	return dstFile, nil
 }
 
+// ParseSnippet parses the AST from incomplete source code snippet.
+func (ap *AstParser) ParseSnippet(source string) ([]dst.Stmt, error) {
+	if source == "" {
+		return nil, ex.New("empty source")
+	}
+	snippet := "package main; func _() {" + source + "}"
+	file, err := decorator.ParseFile(ap.fset, "", snippet, 0)
+	if err != nil {
+		return nil, ex.Wrap(err)
+	}
+	funcDecl, ok := file.Decls[0].(*dst.FuncDecl)
```

Comment:
Is there a chance to trigger the index out of bounds exception?

Reply:
No, because our snippet explicitly defines a function named _, so it's guaranteed to exist.